### PR TITLE
PA-2402, PA-2399 Added Prometheus and Grafana

### PR DESCRIPTION
**Pull request by Fosol**
DevOps configuration and documentation to setup Prometheus and Grafana.

NOTE - Grafana authentication does not currently work.  Need to research Oauth implementation with the Lab and GitHub.

- Cleaned up documentation and templates
- Updated the deployment configuration templates to be more customizable

### CODE REVIEW

File: openshift/4.0/README.md [hunk 1]
```diff
@@ -225,455 +225,30 @@ oc project 354028-dev
 oc process -f pims-db-storage.yaml --param-file=dev.env | oc create --save-config=true -f -
 ```
 
-### Configure Database
+### API Database
```
Comment (Fosol) [inline openshift/4.0/README.md:228]: This readme was getting too long.  Moved sections into their own README files and put them closer to their related templates.

File: openshift/4.0/templates/logging/deploy-elasticsearch.yaml [hunk 2]
```diff
@@ -28,16 +28,44 @@ parameters:
     required: true
     value: "354028"
 
+  - name: IMAGE
```
Comment (Fosol) [inline openshift/4.0/templates/logging/deploy-elasticsearch.yaml:31]: Fixed a few issues with the template and made it more customizable.

File: openshift/4.0/templates/maintenance/README.md [hunk 3]
```diff
@@ -0,0 +1,88 @@
+# Maintenance Web Server Caddy
+
+The proxy caddy provides a way to redirec traffic temporarily during a maintenance period.
```
Comment (Fosol) [inline openshift/4.0/templates/maintenance/README.md:3]: This doc is not technically correct and needs to be updated once I get an opportunity to read and understand Devin's documentation on the Caddy.

File: openshift/4.0/templates/persistent-volume-claims/db-storage.yaml [hunk 4]
```diff
@@ -1,6 +1,32 @@
 kind: Template
 apiVersion: v1
 name: pims-storage-template
+parameters:
```
Comment (Fosol) [inline openshift/4.0/templates/persistent-volume-claims/db-storage.yaml:4]: Moved parameters to top to match consistent templating.

File: openshift/4.0/templates/monitor/deploy-grafana.yaml [hunk 5]
```diff
@@ -0,0 +1,1298 @@
...
+    metadata:
+      name: ${APP_NAME}
+      namespace: ${PROJECT_NAMESPACE}-${ENV_NAME}
+      labels:
+        name: ${APP_NAME}
+        app: ${APP_NAME}
+        role: ${ROLE_NAME}
+        env: ${ENV_NAME}
+      annotations:
+        serviceaccounts.openshift.io/oauth-redirectreference.primary: '{"kind":"OAuthRedirectReference","apiVersion":"v1","reference":{"kind":"Route","name":"${APP_NAME}"}}'
```
Comment (Fosol) [inline openshift/4.0/templates/monitor/deploy-grafana.yaml:1135]: oauth does not currently work.  Need to find out if @goemen had to do additional steps previously to get it working, or whether the Lab still supports it going forward.

**Review by devinleighsmith — APPROVED** [review]

**Pull request merged**